File: 프로그래머스/2/42747. H－Index/H－Index.py
def solution(citations):
    answer = 0
    maxValue = max(citations) + 1
    
    for i in range(maxValue):
        gt = []
        lt = []
        for item in citations:
            if item >= i:
                gt.append(item)
            else:
                lt.append(item)

        if len(gt) >= i and len(lt) <= i:
            answer = i
        
    return answer

File: 프로그래머스/2/42747. H－Index/H－Index.js
function solution(citations) {
    let answer = 0;
    
    for (let i = 0; i <= Math.max(...citations); i++) {
        if (citations.filter((a) => a >= i).length >= i) {
            if (citations.filter((b) => b < i).length <= i) {
                answer = i;
            }
        }
    }
    
    return answer;
}